src/_zkapauthorizer/tahoe.py

```diff
@@ -125,19 +126,21 @@ class NotWriteableError(Exception):
 
 
 @async_retry(_common_tahoe_errors)
-async def upload(
-    client: HTTPClient, inpath: FilePath, api_root: DecodedURL
+async def upload_bytes(
+    client: HTTPClient,
+    data: BytesIO,
```

Comment:
I think this should be `typing.BinaryIO` rather than `BytesIO` (and I think I incorrectly used the latter a few places in this codebase and haven't fixed them yet).